### PR TITLE
Updates to openfast_io to match current dev branch API

### PR DESCRIPTION
**Pull request by mayankchetan**
This PR is NOT ready to merge until TODOs below are cleared.

**Feature or improvement description**

Updates to the `openfast_io` library to match the current `dev` branch API. Once merged, will provide OpenFAST users with an up-to-date library that they can import to read/write OpenFAST files. This follows the `fst_vt` nested dictionary structure developed under `WEIS` and heavily borrows code from there. 

This library is currently limited to `OpenFAST` and does not work with `FAST.Farm` or any of the module drivers.

Related PRs: https://github.com/OpenFAST/openfast/pull/2333, https://github.com/OpenFAST/openfast/pull/2188

**Related issue, if one exists**

https://github.com/OpenFAST/openfast/issues/2233, https://github.com/OpenFAST/openfast/issues/2261, https://github.com/OpenFAST/openfast/issues/1851

**Impacted areas of the software**

openfast_io

**Additional supporting information**

**Test results, if applicable**

`pytest` added to span most OpenFAST input files (drivers not included). Based on @andrew-platt comments in https://github.com/OpenFAST/openfast/issues/2233, need to introduce automated testing.

TODO:

- [x] Add test automations & GH actions
- [ ] Create an OpenFAST [PyPI](https://pypi.org/) account so the openfast_io python package can be installed with pip
- [x] Set up a [trusted publisher](https://docs.pypi.org/trusted-publishers/) to securely upload the python package to PyPI
- [ ] Setup conga-forge for `openfast_io` since most NREL tools like `WEIS` and `WISDEM` prefer it.
- [x] Check if we can eliminate the `Poetry` file ~~We will continue using `Poetry`~~
- [x] Add additional tests based on recommendations in this PR if any
- [x] Address `old API` warning thrown by AD15
- [x] Add simple documentation to help users get started
- [ ] Add badges for PyPi and conda-forge

APIs to upgrade before merge:
- [ ] ~~https://github.com/OpenFAST/openfast/pull/986~~
- [x] https://github.com/OpenFAST/openfast/pull/1295
- [x] https://github.com/OpenFAST/openfast/pull/2203
- [x] https://github.com/OpenFAST/openfast/pull/2358
- [x] https://github.com/OpenFAST/openfast/issues/93
- [x] https://github.com/OpenFAST/openfast/pull/2401
- [x] https://github.com/OpenFAST/openfast/pull/2561

### CODE REVIEW

**Comment by andrew-platt** [comment]
I haven't checked through the code, but just noticed there is a new input in the _AeroDyn_ input file that wasn't included in any of the regression tests yet (adding that line to all AD input files in PR #1295).

- [ ] `3               IntegrationMethod  - Switch to indicate which integration method UA uses (1=RK4, 2=AB4, 3=ABM4, 4=BDF2)`

File: openfast_io/openfast_io/FAST_writer.py [hunk 1]
```diff
@@ -786,8 +786,8 @@ def write_AeroDyn15(self):
             self.write_AeroDyn15Coord()
         
         if self.fst_vt['AeroDyn15']['Wake_Mod'] == 3:
-            if self.fst_vt['AeroDyn15']['UAMod'] == 2:
-                raise Exception('OLAF is called with unsteady airfoil aerodynamics, but OLAF currently only supports UAMod == 1') #TODO: need to check if this holds true now
+            if self.fst_vt['AeroDyn15']['UA_Mod'] == 2:
+                raise Exception('OLAF is called with unsteady airfoil aerodynamics, but OLAF currently only supports UA_Mod == 1') #TODO: need to check if this holds true now
```
Comment (bjonkman) [inline openfast_io/openfast_io/FAST_writer.py:790]: I haven't read much of this code, but I don't understand this comment.  The UnsteadyAero module doesn't allow `UA_Mod` to be 1, including when it is called from OLAF.

Comment (mayankchetan) [reply openfast_io/openfast_io/FAST_writer.py:790]: Thank for catching that! I think I missed that with new AD15 API updates.

Comment (mayankchetan) [reply openfast_io/openfast_io/FAST_writer.py:790]: AeroDyn*

**Comment by mayankchetan** [comment]
Ready for merge @andrew-platt, please squash commits :)

**Pull request merged**